Tella/Scenes/Audio/AudioPlayer/Views/EditAudioView.swift

```diff
@@ -0,0 +1,246 @@
+//
+//  EditAudioView.swift
+//  Tella
+//
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+import SwiftUI
+
+
+
+struct EditAudioView: View {
+    @EnvironmentObject var sheetManager: SheetManager
+    
+    @StateObject var editAudioViewModel: EditAudioViewModel
+    @Binding var isPresented : Bool
+    @State var isBottomSheetShown : Bool = false
+    let kTrimViewWidth = 340.0
+    
+    @State var trailingGestureValue: Double = 0.0
+    @State var leadingGestureValue: Double = 0.0
+    @State var shouldStopLeftScroll = false
+    @State var shouldStopRightScroll = false
+    
+    var body: some View {
+        ZStack {
+            VStack {
+                headerView
+                timeLabelsView
+                trimView
+                displayTimeLabels
+                controlButtonsView
+                Spacer()
+            }
+            EditFileCancelBottomSheet(isShown: $isBottomSheetShown, saveAction: { editAudioViewModel.trimAudio() })
+            if case .error(let message) = editAudioViewModel.trimState{
+                ToastView(message: message)
+            }
+        }
+        .onAppear {
+            editAudioViewModel.onAppear()
+            trailingGestureValue = kTrimViewWidth
+        }
+        .onReceive(editAudioViewModel.$trimState) { value in
+            if value == .loaded(true) {
+                self.isPresented = false
+                self.sheetManager.hide()
+                Toast.displayToast(message: LocalizableVault.editFileSavedToast.localized)
+            }
+        }
+        
+        .navigationBarHidden(true)
+        .background(Color.black.ignoresSafeArea())
+    }
+    
+    
+    var trimView: some View {
+        VStack {
+            ZStack(alignment: .trailing) {
+                ZStack(alignment: .leading) {
+                    Image("edit.audio.soundwaves")
```

Comment:
maybe you can move this code to a helper function that return the background view ?

                    Image("edit.audio.soundwaves")
                        .resizable()
                        .frame(height: 180)
                        .background(Color.white.opacity(0.08))

                    Rectangle().fill(Styles.Colors.yellow.opacity(0.16))
                        .offset(x: leadingGestureValue)
                        .frame(width: abs(leadingGestureValue - trailingGestureValue ), height: 180 )

Reply:
Maybe this will lead to a lot of sub methods, I added some comments on the code to make it more readable ( it is already in a small function trimView